src/utils/postCairoWrite.ts

```diff
@@ -1,18 +1,137 @@
 import assert from 'assert';
 import { createHash } from 'crypto';
 import { readFileSync, writeFileSync } from 'fs';
-import { join } from 'path';
+import * as path from 'path';
+import { compileCairo } from '../starknetCli';
+import { CLIError } from './errors';
+import { callClassHashScript } from './utils';
 
 export const HASH_SIZE = 16;
 export const HASH_OPTION = 'sha256';
 
+/**
+  Is used post transpilation to insert the class hash for any contract that can deploy another.
+  During transpilation 0 is placed where the class hash would be. Now that all the needed files are
+  transpiled then we can calculate the class hashes of each and insert them into the file.
+  @param cairoFilePath: The path to the cairo File being processed.
+  @param pathPrefix: The prefix proceeding the the path to the file
+  @param contractHashToClassHash: A mapping that holds the contract path with out the pathPrefix and maps
+  it to the contracts class hash.
+ */
+export function postProcessCairoFile(
+  cairoFilePath: string,
+  pathPrefix = 'warp_output',
```

Comment:
We should remove this default value, I feel this sort of thing might catch us out later.
Especially since the decision of the default directory is decided by the cli part of the code

Reply:
Yeah good spot!